lib/Sema/CSSimplify.cpp

```diff
@@ -12114,10 +12116,17 @@ ConstraintSystem::simplifyUnresolvedMemberChainBaseConstraint(
 
     auto *memberRef = findResolvedMemberRef(memberLoc);
     if (memberRef && (memberRef->isStatic() || isa<TypeAliasDecl>(memberRef))) {
-      return simplifyConformsToConstraint(
-          resultTy, baseTy, ConstraintKind::ConformsTo,
-          getConstraintLocator(memberLoc, ConstraintLocator::MemberRefBase),
-          flags);
+      auto layout = baseTy->getExistentialLayout();
```

Comment:
Might be worth asserting that the layout doesn't have a superclass here since CSBindings will never include that and we don't check it here